zebrad/src/components/health.rs

```diff
@@ -0,0 +1,115 @@
+//! A simple HTTP health endpoint for Zebra.
+
+use std::net::SocketAddr;
+use std::convert::Infallible;
+
+use abscissa_core::{Component, FrameworkError};
+use hyper::body::Incoming;
+use hyper::server::conn::http1;
+use hyper::service::service_fn;
+use hyper::{Request, Response, StatusCode};
+use serde::{Deserialize, Serialize};
+use tracing::{error, info};
+
+/// Abscissa component which runs a health endpoint.
+#[derive(Debug, Component)]
+#[cfg(feature = "health-endpoint")]
+pub struct HealthEndpoint {}
+
+#[cfg(feature = "health-endpoint")]
+impl HealthEndpoint {
+    /// Create the component.
+    pub fn new(config: &Config) -> Result<Self, FrameworkError> {
+        if let Some(addr) = config.endpoint_addr {
+            info!("Trying to open health endpoint at {}...", addr);
+
+            // Start the health endpoint server in a separate thread to avoid Tokio runtime issues
+            std::thread::spawn(move || {
+                let rt = tokio::runtime::Runtime::new().unwrap();
```

Comment:
```suggestion
                 let rt = tokio::runtime::Runtime::new().expect("Tokio runtime should be available");
```
Consider replacing `unwrap()` with `expect(...)`